### PR TITLE
Create story on auto-merge failure

### PR DESCRIPTION
**Pull request by emosbaugh**
This will create a story on Dependabot PR workflow failures for both build-test failures as well as when auto-merge conditions are not satisfied.

### CODE REVIEW

File: .github/workflows/dependabot-automerge.yaml [hunk 1]
```diff
@@ -60,6 +60,44 @@ jobs:
...
 
+      - name: Create shortcut story if auto-merge conditions are not satisfied
+        if: ${{ steps.auto-approve-conditions.conclusion != 'success' }}
+        run: |
+          EXISTING_STORY="$(curl -X GET \
+            -H "Content-Type: application/json" \
+            -H "Shortcut-Token: ${SHORTCUT_API_TOKEN}" \
+            -d "{ \"page_size\": 1, \"query\": \"title:\\\"Automerge Failed for #${PR_NUMBER}\\\"\" }" \
+            -L "https://api.app.shortcut.com/api/v3/search/stories")"
+          if echo "${EXISTING_STORY}" | grep -q "Automerge Failed for #${PR_NUMBER} " ; then
```
Comment (emosbaugh) [inline .github/workflows/dependabot-automerge.yaml:71]: i think im gonna explore changing this to a comment with ch story and looking it up from there

File: .github/workflows/dependabot-automerge.yaml [hunk 2]
```diff
@@ -60,6 +60,44 @@ jobs:
...
+              \"estimate\": 1 }" \
+            -L "https://api.app.shortcut.com/api/v3/stories"
+
+          echo "::notice ::shortcut story created"
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          SHORTCUT_API_TOKEN: ${{ secrets.SHORTCUT_API_TOKEN }}
+          GITHUB_REPO: ${{ github.repository }}
+          PR_NUMBER: ${{ github.event.pull_request.number }}
+          PR_URL: ${{ github.event.pull_request.html_url }}
```
Comment (DanStough) [inline .github/workflows/dependabot-automerge.yaml:97]: I don't think this is used in this step.

Comment (DanStough) [inline .github/workflows/dependabot-automerge.yaml:97]: I don't think this is used in this step.

File: .github/workflows/dependabot-automerge.yaml (hunk 1 above)
Comment (DanStough) [inline .github/workflows/dependabot-automerge.yaml:71]: Is the trailing space here going to screw up the search? I don't see it in the create.

**Review by DanStough — DISMISSED** [review]
Couple minor things

**Review by DanStough — APPROVED** [review]

**Pull request merged**